python/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
+
+# TODO: Pull this directly from the file written by rapids_cmake_write_version_file.
+set(rmm_version 22.04.00)
+
+project(
+  rmm-python
+  VERSION ${rmm_version}
+  LANGUAGES # TODO: Building Python extension modules via the python_extension_module requires the C
+            # language to be enabled here. The test project that is built in scikit-build to verify
+            # various linking options for the python library is hardcoded to build with C, so until
+            # that is fixed we need to keep C.
+            C
+            CXX
+            # TODO: The C++ RMM CMake configuration targets cuda_std_17 features, which prior to
+            # CMake 3.22 will also pull in the corresponding required languages even if this project
+            # does not actually require those languages. As a result, we need to include CUDA here.
+            # We can remove CUDA once we upgrade the minimum required CMake version to 3.22.
+            CUDA)
+
+option(FIND_RMM_CPP "Search for existing RMM C++ installations before defaulting to local files"
+       OFF)
+
+find_package(PythonExtensions REQUIRED)
+find_package(Cython REQUIRED)
+
+# TODO: Should we guard this based on a scikit-build version? Override this function to avoid
+# scikit-build clobbering symbol visibility.
+function(_set_python_extension_symbol_visibility _target)
+  if(PYTHON_VERSION_MAJOR VERSION_GREATER 2)
+    set(_modinit_prefix "PyInit_")
+  else()
+    set(_modinit_prefix "init")
+  endif()
+  message("_modinit_prefix:${_modinit_prefix}")
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    set_target_properties(${_target} PROPERTIES LINK_FLAGS "/EXPORT:${_modinit_prefix}${_target}")
+  elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(_script_path ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${_target}-version-script.map)
+    file(
+      WRITE ${_script_path}
+      # Note: The change is to this script, which does not indiscriminately mark all non PyInit
+      # symbols as local.
+      "{global: ${_modinit_prefix}${_target}; };")
+    set_property(
+      TARGET ${_target}
+      APPEND_STRING
+      PROPERTY LINK_FLAGS " -Wl,--version-script=\"${_script_path}\"")
+  endif()
+endfunction()
+
+# Ignore unused variable warning.
+set(ignored_variable "${SKBUILD}")
+
+# If the user requested it we attempt to find RMM. TODO: Should we allow the user to specify a path
+# instead of just searching? This version assumes that the installation has been appropriately
+# configured for CMake discovery.
+if(FIND_RMM_CPP)
+  find_package(RMM ${rmm_version})
```

Comment:
By default with a `find_package` call a user can provide an `RMM_ROOT` CMake or environment variable to influence where it searches to find the package. If I remember correctly we did the work to make sure that an rmm build directory could suffice for this call, but I could be mistaken.

Reply:
When you say that we did the work, are you referring to your original scikit-build PR or the current RMM CMake setting the appropriate environment variables to make itself discoverable?

Reply:
current librmm can be found via the build directory. librmm doesn't need to do anything, the call to rmm would specify one of the following:

- `RMM_ROOT` ( env var or CMake var )
- `CMAKE_PREFIX_PATH` ( env var or CMake var )